src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -335,6 +336,14 @@ private void EventLoop()
                             }
                         }
                     }
+
+                    // HandleEvents queues to the ThreadPool; so we queue the next read 
+                    // after all the events have been picked up by the ThreadPool (but not necessarily complete)
+                    // This makes it more likely there are events and the batches are larger (less syscalls)
+                    // as well as not overloading the system with events at a higher rate than they can actually be 
+                    // processed, which breaks any back pressure on the sockets.
+                    ThreadPool.UnsafeQueueUserWorkItem((mres) => ((ManualResetEventSlim)mres).Set(), _batchProcessed);
+                    _batchProcessed.Wait();
```

Comment:
You're never resetting _batchProcessed, and this is allocating a work item on every queueing.

Reply:
In theory I can see how this could help when lots of work arrives at the same time.  When it doesn't, however, this will make individual socket operations more expensive.

How are you aiming to validate this?

Reply:
> You're never resetting _batchProcessed, so as currently written it's never blocking after the first batch, and this is allocating a work item on every queueing.

🤦‍♂

Reply:
> How are you aiming to validate this?

Not sure how to open VS with linux source rather than Windows; so probably slowly...

Reply:
> Not sure how to open VS with linux source rather than Windows

Right-click on the solution, select Configuration Manager, and for the src project, change its configuration to `netcoreapp-Unix-Debug`.  You may need to save and then reload the project for everything to work correctly, though it's possible those issues have been fixed.

That said, I actually meant more around perf validation :)